src/components/example/index.tsx

```diff
@@ -40,7 +43,40 @@ const CertificatesTable: React.FC = () => {
     };
     fetchCertificates();
   }, []);
+  const renderActions = (id?: number) => (
+    <div className="">
+      <img
+        className="setting-icon"
+        width={20}
+        height={20}
+        src={SettingsIcon}
+        alt="Setting icon"
+        onClick={() => setIsOpen(isOpen === id ? undefined : id)}
+      />
+      {isOpen === id && (
+        <div ref={popupRef} className="setting-action-buttons">
+          <Button
+            onClick={() => navigate(`${routes.example1.url}/${id}`)}
+            label="Edit"
+          />
+          <Button label="Delete" />
+        </div>
+      )}
+    </div>
+  );
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        popupRef.current &&
+        !popupRef.current.contains(event.target as Node)
+      ) {
+        setIsOpen(undefined);
+      }
+    };
```

Comment:
### Feedback:
- Transform code for `handleClickOutside` into a reusable custom hook called useClickOutside. Here's how you can do it:
```JSX
import { useEffect, useRef, RefObject } from 'react';

function useClickOutside<T extends HTMLElement = HTMLElement>(
  handler: () => void
): RefObject<T> {
  const ref = useRef<T>(null);

  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (ref.current && !ref.current.contains(event.target as Node)) {
        handler();
      }
    };

    document.addEventListener('mousedown', handleClickOutside);

    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, [handler]);

  return ref;
}
```
- `Create a Separate Component:`
Extract the action rendering logic into its own component. This will improve reusability and maintainability.
```JSX
const ActionMenu = ({ id, onEdit, onDelete }) => {
  const [isOpen, setIsOpen] = useState(false);
  const popupRef = useClickOutside(() => setIsOpen(false));
  
  return (
    <div className="">
      <img
        className="setting-icon"
        width={20}
        height={20}
        src={SettingsIcon}
        alt="Setting icon"
        onClick={() => setIsOpen(!isOpen)}
      />
      {isOpen && (
        <div ref={popupRef} className="setting-action-buttons">
          <Button onClick={onEdit} label="Edit" />
          <Button onClick={onDelete} label="Delete" />
        </div>
      )}
    </div>
  );
};
```

### Benefits
- Improved reusability across different parts of the application
- Better separation of concerns
- More flexible and customizable component
- Easier to test individual parts